cross-doc-explainer.md

```diff
@@ -0,0 +1,114 @@
+
+
+### Issues
+
+1. The meta tag can be used to opt-in to other navigation types going forward: same-document, same-site, etc.
+
+1. Using meta-tags prevents the declaration being controlled by media queries, which feels important for `prefers-reduced-motion`.
+See [w3c/csswg-drafts#8048](https://github.com/w3c/csswg-drafts/issues/8048).
+
+## Lifecycle
+
+![Lifecycle chart for cross-document transitions](media/mpa-chart.svg)
+
+### Capturing the old state
+
+The pold state is captured](https://drafts.csswg.org/css-view-transitions-1/#capture-old-state-algorithm) right before the old document is hidden and a new one is shown.
+In the HTML spec that moment is defined [here](https://html.spec.whatwg.org/#populating-a-session-history-entry:loading-a-document).
+This can either happen during normal navigations, when the new document is about to be created,
+in Back/Forward cache navigations, or when activating a prerendered document.
+
+Before creating the new document (or activating a cached/prerendered one), the UA would update the rendering and snapshot the old document, in the same manner a document is snapshotted for a same-document navigation.
+
+Note that currently there are no planned new events for the exit transition.
+The developer can use existing events like `navigate` or `click` to customize the
+exit transition at moments that should be late enough.
+
+#### Issues
+
+1. Should we enable skipping the transition on exit? This could be done on`pagehide`.
+
+### Capturing the new state
+
+The [new state is captured](https://drafts.csswg.org/css-view-transitions-1/#capture-new-state-algorithm) at the first [rendering opportunity](https://html.spec.whatwg.org/#rendering-opportunity)
+of the new document. This allows the new document to use the
+[render-blocking mechanism](https://html.spec.whatwg.org/#render-blocking-mechanism) as a way to
+delay the transition.
+
+For most cases, opting in via the meta tag, styling with the pseudo-elements, and delaying the transition using the
+[render-blocking mechanism](https://html.spec.whatwg.org/#render-blocking-mechanism) should be sufficient. But there are certain cases where further customization is desired, for example:
+
+* Abort the transition if certain conditions apply.
+
+* Modify the pseudo-element style based on certain conditions, e.g. apply transition
+   names only to images that are already loaded.
+
+* Prepare the exit transition right before capture.
+
+There are several options as to how to enable this:
+
+1. Fire an event right before the new document capture. This would be either right
+   before the first `requestAnimationFrame` callback, or after `pageshow` in the reactivation case. Potential names: `reveal`, `beforepagetransition`. Note that in the latter case, it's too late to delay the transition via the render-blocking mechanism. This event doesn't have to be specific to transitions.
+
+1. Expose something like a `document.currentRevealTransition` or so that the script can
```

Comment:
I don't think its about choosing between one of these options, you need both. First one for customizing the animation based on what was fetched before first paint. And second one for customizing what is render blocking.